### PR TITLE
[Merged by Bors] - feat: add lemma `Finsupp.erase_apply`

### PR DESCRIPTION
**Pull request by BoltonBailey**
Adds a lemma

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Finsupp/Defs.lean [hunk 1]
```diff
@@ -661,6 +661,14 @@ theorem erase_ne {a a' : α} {f : α →₀ M} (h : a' ≠ a) : (f.erase a) a' =
   classical simp only [erase, coe_mk, h, ite_false]
 #align finsupp.erase_ne Finsupp.erase_ne
 
+theorem erase_apply [DecidableEq α] {a a' : α} {f : α →₀ M} :
+    (f.erase a) a' = if a' = a then 0 else f a' := by
+  classical
+  split_ifs with h
+  · rw [h]
+    exact erase_same
+  · exact erase_ne h
```
Comment (eric-wieser) [inline Mathlib/Data/Finsupp/Defs.lean:670]:
```suggestion
  rw [erase, coe_mk]
  convert rfl
```
works too

File: Mathlib/Data/Finsupp/Defs.lean [hunk 2]
```diff
@@ -661,6 +661,14 @@ theorem erase_ne {a a' : α} {f : α →₀ M} (h : a' ≠ a) : (f.erase a) a' =
   classical simp only [erase, coe_mk, h, ite_false]
 #align finsupp.erase_ne Finsupp.erase_ne
 
+theorem erase_apply [DecidableEq α] {a a' : α} {f : α →₀ M} :
```
Comment (eric-wieser) [inline Mathlib/Data/Finsupp/Defs.lean:664]: Arguably these should all be explicit

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Data/Finsupp/Defs.lean (hunk 2 above)
Comment (BoltonBailey) [reply Mathlib/Data/Finsupp/Defs.lean:664]: I tend to agree, but I also see that semantically similar lemmas nearby leave them implicit. I think I'll leave this for a future PR to correct, hope that's ok.

**Pull request closed without merging**